master/internal/api_user.go

```diff
@@ -287,6 +293,9 @@ func (a *apiServer) PostUser(
 	if req.Password != "" && req.User.Remote {
 		return nil, status.Error(codes.InvalidArgument, "cannot set password for remote user")
 	}
+	if req.User.Remote && !version.IsEE {
+		return nil, errUserRemoteRequiresEE
```

Comment:
maybe add intg tests? so someone can't break this

Reply:
yep, got it.

Reply:
well, did something slightly more complicated, so that it is correct in both cases 
```
	tmp := version.IsEE
	version.IsEE = true
	defer func() { version.IsEE = tmp }()
```